### PR TITLE
annotate errors in e2e tests

### LINKED ISSUE #34059: destroy the "Expect(err).NotTo(HaveOccurred())" anti-pattern in e2e tests
I'm getting tired of seeing flake issues like https://github.com/kubernetes/kubernetes/issues/33647 where every single failure is of the form

```
Expected error:
    <*errors.errorString | 0xc820412c10>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred
```

This is about as unhelpful an error message as you can get.

The problem is that we have

``` go
Expect(err).NotTo(HaveOccurred())
```

peppered throughout the code everywhere, and in this form, you can only figure out what failed if you read the Ginkgo error, remember which line has the line number you actually want, and then try to read surrounding code in that file.

Possible improvements given in http://onsi.github.io/gomega/#handling-errors:
- Use `Succeed()`
- Annotate the assertions
- Basically, do anything except just Expect(err).NotTo(HaveOccurred())

cc @kubernetes/sig-testing

### PR DESCRIPTION
**Pull request by phoenixking25**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

#34059 

**Special notes for your reviewer**:
This is my first time writing Go/Contributing to Kubernetes - let me know if I've missed anything! :-)

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
/ok-to-test

**Comment by phoenixking25** [comment]
@feiskyer how to pass this test, getting no idea how it failed from logs

**Comment by phoenixking25** [comment]
i think it can be merged now @feiskyer

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -1835,7 +1835,7 @@ func WaitForEndpoint(c clientset.Interface, ns, name string) error {
 			Logf("Endpoint %s/%s is not ready yet", ns, name)
 			continue
 		}
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "Endpoint %s/%s is not ready yet", ns, name)
```
[the remark this answers is not in the document]
Comment (phoenixking25) [reply test/e2e/framework/util.go]: Ok i will correct that 👍 @oomichi

[the remark this answers is not in the document]
Comment (phoenixking25) [reply test/e2e/framework/util.go]: @oomichi changed line to `"Failed to get endpoints for %s/%s", ns, name`

**Comment by hoegaarden** [comment]
nit: Should we try to be consistent here with upper- vs. lowercase? `failed to get namespace ...` vs .`Failed to get endpoints for ...`

**Comment by phoenixking25** [comment]
@hoegaarden ok will change all to uppercase.

**Comment by hoegaarden** [comment]
@phoenixking25 Could you remove the `Fixes ...` from the PR's message -- I guess it does not remove *all* occurrences of plain `Except(err).NotTo(HaveOccoured())`.

**Comment by phoenixking25** [comment]
@hoegaarden done

**Review by oomichi — COMMENTED** [review]
/lgtm

File: test/e2e_node/density_test.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 	"sync"
 	"time"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (mkumatag) [inline test/e2e_node/density_test.go]: Not really sure why this change is required here!

**Comment by mkumatag** [comment]
Please squash all the commits..

**Comment by ixdy** [comment]
/approve

thanks!
